src/main/java/com/auction/controller/AuctionController.java

```diff
@@ -26,46 +29,40 @@ public List<Auction> getALlAuctions() {
     }
 
     @GetMapping("/{auctionId}")
-    public Auction getAuctionById(@PathVariable int auctionId) {
-        return auctionService.findById(auctionId);
+    public ResponseEntity<Auction> getAuctionById(@PathVariable int auctionId) {
+        Auction auction = auctionService.findById(auctionId);
+        return auction != null ? ResponseEntity.ok(auction) : ResponseEntity.status(HttpStatus.NOT_FOUND).body(null);
     }
 
-    @PostMapping("/addNewAuction/{productId}")
-    public Auction addNewAuction(@RequestBody Auction theAuction, @PathVariable int productId) {
+    @PostMapping("/addNewAuction")
+    public Auction addNewAuction(
+            @RequestParam("productId") int productId,
```

Comment:
if you are using so many params and POST method - it is better to use specific object that will contains all that fields.